### PR TITLE
nix: composable overlays

### PR DESCRIPTION
**Pull request by donn**
## Tool Updates

* Updated nix-eda to `0814aa6`: more orthodox approach to managing dependencies by overlaying them on top of nixpkgs, which fixes an occasional "repeated allocation" issue and helps make override behavior more consistent.
  * Yosys and first-party plugins updated to 0.46 
* OpenLane itself no longer included in `devShells.*.dev`, `devShells.*.docs`
  * These shells are intended to be actual dev shells, i.e. used to develop OpenLane, and needing OpenLane to pass tests to run these shells makes no sense.

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -20,5 +20,6 @@ exclude =
     test/designs/*.py,
     examples,
     sandbox/
```
Comment (kareefardi) [inline .flake8:22]: is a comma needed here?

File: openlane/config/variable.py [hunk 2]
```diff
@@ -447,7 +447,7 @@ def __process(
                     raw = raw.split(";")
                 else:
                     raw = raw.split()
-                if raw[-1] == "":
+                if len(raw) and raw[-1] == "":
```
Comment (kareefardi) [inline openlane/config/variable.py:450]: does this fix a bug?

File: nix/openroad.nix [hunk 3]
```diff
@@ -118,7 +118,7 @@
     nativeBuildInputs = [
       swig4
       pkg-config
-      cmake
+      python3.pkgs.cmake # TODO: Replace with top-level cmake, I'm just doing this to avoid a rebuild
```
Comment (kareefardi) [inline nix/openroad.nix:121]: should this be replaced?

File: .flake8 (hunk 1 above)
Comment (donn) [reply .flake8:22]: I think so yeah, thanks for the catch

File: nix/openroad.nix (hunk 3 above)
Comment (donn) [reply nix/openroad.nix:121]: Not yet

File: openlane/config/variable.py (hunk 2 above)
Comment (donn) [reply openlane/config/variable.py:450]: Yes, though this bug is in main and should be fixed there as well

Comment (kareefardi) [reply openlane/config/variable.py:450]: Then perhaps this should be a separate PR.

**Review by kareefardi — APPROVED** [review]

**Pull request merged**